src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -322,6 +316,55 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXIsInfOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXIsInfOp> {
+  using FOp = CustomScalarOp;
+  using IOp = NotSuportedScalarOp;
+};
+
+template <>
+Value emitScalarOpFor<ONNXIsInfOp>(ConversionPatternRewriter &rewriter,
+    Location loc, Operation *op, Type elementType,
+    ArrayRef<Value> scalarOperands) {
+
+  Value operand = scalarOperands[0];
+  Type inputElemType = getElementType(operand.getType());
+  Value result;
+
+  CheckIfCustomScalarOpIsSupported<ONNXIsInfOp>(inputElemType);
+  MultiDialectBuilder<MathBuilder> create(rewriter, loc);
+  Value negInf = create.math.negativeInf(inputElemType);
+  Value posInf = create.math.positiveInf(inputElemType);
+
+  double detectNegAttribute = dyn_cast<ONNXIsInfOp>(op).getDetectNegative();
+  double detectPosAttribute = dyn_cast<ONNXIsInfOp>(op).getDetectPositive();
+
+  // Three different cases: Infinity, Negative Infinity and Positive Infinity
+  bool detectInf = detectPosAttribute == 1 && detectNegAttribute == 1;
+  bool detectNeg = detectNegAttribute == 1 && detectPosAttribute == 0;
+  bool detectPos = detectPosAttribute == 1 && detectNegAttribute == 0;
+
+  Value equPos = create.math.eq(operand, posInf);
+  Value equNeg = create.math.eq(operand, negInf);
```

Comment:
I have it outside of the `if` to avoid redundancy. I use equPos and equNeg in the first `if` which is to detect regular infinity. So I guess either way there will be redundancy.  It is used here:

```
 if (detectInf) {
    // If infinity return true for both positive and negative infinity
    result = create.math.ori(equPos, equNeg);
  }
```

Reply:
Okay I updated this...let me know your thoughts.